### DISCUSSION TITLE
Find runs-on based on check_run event

### DISCUSSION THREAD

**Discussion by npalm** [actions/runner#1166] [Q&A]
I would like to find the run-on tag for a specific job based on the check_run event. The check_run event contains the job_name and check_suite_Id. With this information I can find the run-on but it requires potential to search for all workflow runs.

- based on the check_run event I can find the the check_suite_id and job_name using the check_run API call
- next I can query using the actions api for all runs
- next I can filter based on the check_suite_id the and job_name the matching run
- with this information I can find the workflow file and also the run-on field.

Does anybody knows if it possible to find the right workflow without searching trough all runs?

**Comment by TingluoHuang** [comment]
We are adding a new workflow job webhook event which will include the required runner labels for each job, maybe you can get benefit from that?

**Reply by npalm** [reply]
@TingluoHuang sounds good, when do you expect the new event will be releases? Is there an issue that I can track?

**Reply by npalm** [reply]
@TingluoHuang kind reminder, can you let us know when we can expect the new event?
